test/src/tests-batchUploader.ts

```diff
@@ -249,7 +248,7 @@ describe('batch uploader', () => {
             window.mParticle.config.identityCallback = function(result) {
                 let currentUser = result.getUser()
                 if (currentUser) {
-                    currentUser.setUserAttribute("number", Math.floor((Math.random() * 1000) + 1))
+                    currentUser.setUserAttribute("number", `${Math.floor((Math.random() * 1000) + 1)}`)
```

Comment:
not sure why this was changed or if some test was failing, but I don't think this is necessary. I think the idea was we are setting a user attribute to an actual number, not a string of a number

Reply:
Ah, very interesting. Seems like from the [docs](https://docs.mparticle.com/developers/sdk/web/users/#set-user-attributes) for `setUserAttribute` we allow `string` and `null`, and in the case of `setUserAttributeList` we allow `array`, but the `setUserAttribute` code shows we run this [validation code](https://github.com/mParticle/mparticle-web-sdk/blob/master/src/validators.ts#L15-L17), which _does allow_ `numbers` and `booleans in addition to `strings`.  I think for now this is fine, and I'm going to make a ticket to sort out how this works on the other platforms.

Reply:
i dont have much context here but setting a user attribute to a `number` seems like a valid case. if it is, then the type is wrong. if its not, then the test is _wrong_

Reply:
Yep it absolutely is from a user perspective. I'll add some more context based on my research yesterday.  I added a sustained engineering issue yesterday at go.mparticle.om/SQDSDKS-4576 to track this and summarize the above. I should have referenced it in my comment.

When the client sends a number to the server, the server casts it to a `string` for some reason, which is probably why I _think_ iOS/Android only allows `string`/`null` to not mislead the customer into thinking it's saved as a `number`.  Our SDK does allow numbers (but our types in definitely typed _don't_), so that's causing the issue here. 

We should _probably_ match iOS/Android, and decide what to do about definitely typed. Since DT is potentially wrong, we should match it, but perhaps add a comment to the go link above.
```suggestion
                    // TODO: Investigate if we should update definitely typed typings for setUserAttribute which only allows strings right now - more context at go.mparticle.om/SQDSDKS-4576
                    currentUser.setUserAttribute("number", `${Math.floor((Math.random() * 1000) + 1)}`)
```